types/any_object_holder.go

```diff
@@ -0,0 +1,143 @@
+package types
+
+import (
+	"fmt"
+	"strings"
+)
+
+// HoldableObject defines a common interface for types which can be placed in AnyObjectHolder
+type HoldableObject interface {
+	RVType
+	ObjectID() RVType // Returns the object identifier of the type
+}
+
+// AnyObjectHolderObjects holds a mapping of RVTypes that are accessible in a AnyDataHolder
+var AnyObjectHolderObjects = make(map[RVType]HoldableObject)
+
+// RegisterObjectHolderType registers a RVType to be accessible in a AnyDataHolder
+func RegisterObjectHolderType(rvType HoldableObject) {
+	AnyObjectHolderObjects[rvType.ObjectID()] = rvType
+}
+
+// AnyObjectHolder can hold a reference to any RVType which can be held
+type AnyObjectHolder[T HoldableObject] struct {
+	Object T
+}
+
+// WriteTo writes the AnyObjectHolder to the given writable
+func (aoh AnyObjectHolder[T]) WriteTo(writable Writable) {
+	contentWritable := writable.CopyNew()
+
+	aoh.Object.WriteTo(contentWritable)
+
+	objectBuffer := NewBuffer(contentWritable.Bytes())
+
+	objectBufferLength := uint32(len(objectBuffer) + 4) // * Length of the Buffer
+
+	aoh.Object.ObjectID().WriteTo(writable)
+	writable.WriteUInt32LE(objectBufferLength)
+	objectBuffer.WriteTo(writable)
+}
+
+// ExtractFrom extracts the AnyObjectHolder from the given readable
+func (aoh *AnyObjectHolder[T]) ExtractFrom(readable Readable) error {
+	var err error
+
+	// TODO - This assumes the identifier is a String
+	identifier := NewString("")
+	err = identifier.ExtractFrom(readable)
+	if err != nil {
+		return fmt.Errorf("Failed to read AnyObjectHolder identifier. %s", err.Error())
+	}
+
+	length := NewUInt32(0)
+	err = length.ExtractFrom(readable)
+	if err != nil {
+		return fmt.Errorf("Failed to read AnyObjectHolder length. %s", err.Error())
+	}
+
+	// * This is technically a Buffer, but we can't instantiate a new Readable from here so interpret it as a UInt32 and the object data
+	bufferLength := NewUInt32(0)
+	err = bufferLength.ExtractFrom(readable)
+	if err != nil {
+		return fmt.Errorf("Failed to read AnyObjectHolder buffer length. %s", err.Error())
+	}
+
+	if _, ok := AnyObjectHolderObjects[identifier]; !ok {
+		return fmt.Errorf("Unknown AnyObjectHolder identifier: %s", identifier)
+	}
+
+	ptr := AnyObjectHolderObjects[identifier].CopyRef()
+
+	if err := ptr.ExtractFrom(readable); err != nil {
+		return fmt.Errorf("Failed to read AnyObjectHolder object. %s", err.Error())
+	}
+
+	var ok bool
+	aoh.Object, ok = ptr.Deref().(T)
+	if (!ok) {
+		return fmt.Errorf("Input AnyObjectHolder object %s is invalid", identifier)
+	}
```

Comment:
This can be slightly simplified down to:

```suggestion
	var ok bool
	if aoh.Object, ok = ptr.Deref().(T); !ok {
		return fmt.Errorf("Input AnyObjectHolder object %s is invalid", identifier)
	}
```

Though this is not strictly necessary